app/components/dynamic-scan/expiry/index.ts

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable ember/no-computed-properties-in-native-classes, ember/no-observers */
+import Component from '@glimmer/component';
+import { action, computed } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+import { later } from '@ember/runloop';
+import { task } from 'ember-concurrency';
+import dayjs from 'dayjs';
+import { addObserver, removeObserver } from '@ember/object/observers';
+import Store from '@ember-data/store';
+import FileModel from 'irene/models/file';
+import DatetimeService from 'irene/services/datetime';
+import { Duration } from 'dayjs/plugin/duration';
+import { EmberRunTimer } from '@ember/runloop/types';
+import DynamicscanModal from 'irene/models/dynamicscan';
+import ENV from 'irene/config/environment';
+
+export interface DyanmicScanExpirySignature {
+  Args: {
+    file: FileModel;
+  };
+}
+
+export default class DyanmicScanExpiryComponent extends Component<DyanmicScanExpirySignature> {
+  @service('notifications') declare notify: NotificationService;
+  @service declare store: Store;
+  @service declare datetime: DatetimeService;
+
+  @tracked dynamicscan: DynamicscanModal | null = null;
+  @tracked durationRemaining: null | Duration = null;
+  @tracked clockStop = false;
+  @tracked extendBtnAnchorRef: HTMLElement | null = null;
+
+  constructor(owner: unknown, args: DyanmicScanExpirySignature['Args']) {
+    super(owner, args);
+
+    this.fetchDynaminscan.perform();
+
+    this.clock();
+
+    addObserver(this.args.file, 'isReady', this.observeDeviceState);
+  }
+
+  willDestroy() {
+    super.willDestroy();
+
+    this.clockStop = true;
+
+    removeObserver(this.args.file, 'isReady', this.observeDeviceState);
+  }
+
+  get extendTimeOptions() {
+    return [5, 15, 30];
+  }
+
+  fetchDynaminscan = task(async () => {
+    const id = this.args.file.id;
+    this.dynamicscan = await this.store.findRecord('dynamicscan', id);
+  });
+
+  observeDeviceState() {
+    this.fetchDynaminscan.perform();
+  }
+
+  @computed('durationRemaining')
```

Comment:
Are these computed decorators required since we're using getters here? Same with the others used in this class.

Reply:
This has been removed